Library/Homebrew/test/dev-cmd/bottle_spec.rb

```diff
@@ -1,4 +1,11 @@
 describe "brew bottle", :integration_test do
+  before do
+    if OS.linux?
+      setup_test_formula "patchelf"
```

Comment:
`stub_formula_loader formula("patchelf")` or `stub_formula_loader formula("patchelf") { url "patchelf-1.0" }` would be preferable, I think. They could also be done unconditionally. The `installed?` check can also be mocked or stubbed using https://github.com/rspec/rspec-mocks#method-stubs

Reply:
I've used `stub_formula_loader` and mocked the `installed?` method for the other two tests `test/formula_installer_bottle_spec.rb` and `test/formulary_spec.rb`. That approach did not work here in `test/dev-cmd/bottle_spec.rb`. Because this test is an integration test and shells out to `brew`, mocking the methods do not appear to survive through the shelling out.

Replacing these two lines
```ruby
      setup_test_formula "patchelf"
      (HOMEBREW_CELLAR/"patchelf/1.0/bin").mkpath
```
with
```ruby
      stub_formula_loader formula("patchelf") { url "patchelf-1.0" }
      allow(Formula["patchelf"]).to receive(:installed?).and_return(true)
```
resulted in the error:
```
  1) brew bottle builds a bottle for the given Formula
     Failure/Error:
       expect { brew "bottle", "--no-rebuild", testball }
         .to output(/Formula not from core or any taps/).to_stderr
         .and not_to_output.to_stdout
         .and be_a_failure
     
       expected block to output /Formula not from core or any taps/ to stderr, but output "Error: No available formula with the name \"patchelf\" \n"
     # ./test/dev-cmd/bottle_spec.rb:12:in `block (2 levels) in <top (required)>'
     # ./test/support/helper/spec/shared_context/integration_test.rb:44:in `block (2 levels) in <top (required)>'
     # ./test/spec_helper.rb:117:in `block (2 levels) in <top (required)>'
     # ./vendor/bundle/ruby/2.3.0/gems/rspec-retry-0.6.1/lib/rspec/retry.rb:123:in `block in run'
```